gensim/test/test_corpora_dictionary.py

```diff
@@ -309,6 +309,16 @@ def test_dict_interface(self):
             self.assertTrue(isinstance(d.keys(), list))
             self.assertTrue(isinstance(d.values(), list))
 
+    def test_patch_with_special_tokens(self):
+        special_tokens = {'pad': 0, 'space': 1}
+        corpus = [["máma", "mele", "maso"], ["ema", "má", "máma"]]
+        d = Dictionary(corpus)
+        self.assertEqual(len(d.token2id), 5)
+        d.patch_with_special_tokens(special_tokens)
+        self.assertEqual(d.token2id['pad'], 0)
+        self.assertEqual(d.token2id['space'], 1)
+        self.assertEqual(len(d.token2id), 7)
```

Comment:
Need to extend this test (check also than you can transform document with special tokens)

Reply:
Extension added.

Reply:
It wasn't obvious to me that this was the indention - to test the doc2bow transformation. I have added tests for that now, however.

Reply:
That's strange because this is the main method for an apply dictionary to document to get BoW, thanks!